### PR TITLE
[B2BP-422] - Add logo to Header

### PR DESCRIPTION
**Pull request by Meolocious**
#### List of Changes
- Updated EC version from 3.0.0 to 3.1.0
- Added logo to Header
- Updated Header colours
- Added popup shadow

#### Motivation and Context
The reason of this feat is to align header component to SEND legacy

#### How Has This Been Tested?
Tested in local

#### Types of changes
- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] Doesn't need any documentation updates

### CODE REVIEW

File: apps/nextjs-website/src/components/Header.tsx [hunk 1]
```diff
@@ -3,28 +3,53 @@ import { Header as HeaderEC } from '@pagopa/pagopa-editorial-components/dist/com
...
+    };
+  }
+  // allow configuration using `createTheme`
+  interface ThemeOptions {
+    custom?: {
+      boxShadow?: string;
+    };
+  }
+}
+
```
Comment (kin0992) [inline apps/nextjs-website/src/components/Header.tsx:23]: why is this needed?

Comment (Meolocious) [reply apps/nextjs-website/src/components/Header.tsx:23]: without that part, this line here

`boxShadow: theme.custom.boxShadow,`

will have this error here: Property 'custom' does not exist on type 'Theme'

due to not use hardcoded value inside file, I proceeded to create the theme.ts file to use it for custom colour value and I didn't find another way to do this. I'm not a fan of declaring that part inside header.tsx but I didn't found another way

Comment (simosalsi) [reply apps/nextjs-website/src/components/Header.tsx:23]: You should be able to use theme-aware values.

I'm making a quick suggestion to show the changes.

Comment (simosalsi) [inline apps/nextjs-website/src/components/Header.tsx:23]:
```suggestion
```
Remove if using theme-aware values.

File: apps/nextjs-website/src/components/Header.tsx [hunk 2]
```diff
@@ -37,20 +62,28 @@ const Header = (props: HeaderWithNavigation) => {
...
-            '.MuiStack-root': {
-              // Popup on hover
+            '.MuiTypography-root': {
+              textDecoration: 'none',
+              fontWeight: 600,
+              fontSize: '1rem',
+            },
+            '& .MuiStack-root': {
               zIndex: 10,
+              boxShadow: theme.custom.boxShadow,
```
Comment (simosalsi) [inline apps/nextjs-website/src/components/Header.tsx:80]:
```suggestion
              boxShadow: 'custom.boxShadow',
```
Use theme-aware values.

**Review by simosalsi — CHANGES_REQUESTED** [review]

**Review by kin0992 — APPROVED** [review]
